banner formation

lucesitaliss opened the pull request:
## Photo or video showing UI changes (if any)

## Description

A few sentences describing the overall goals of the pull request's commits. Include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update.

# Keep this guide in mind to review and auto-review PR code.

## Nomenclatura y convenciones.

- [ ] Use camelCase to name variables and functions
- [ ] Use PascalCase to name components in React
- [ ] Use descriptive and consistent names for all variables, functions, components and files. DO NOT USE A SINGLE LETTER TO NAME VARIABLES. eg: `error => e` , `index => i`
- [ ] Use eslint and prettier to keep bug cleanup. DO NOT LEAVE ESLINT WARNINGS UNSOLVED
- [ ] Vary all numbers and text strings. DO NOT ENTER MAGIC STRINGS, OR MAGIC NUMBERS e.g.:

  `'template_discount'` => `ACTION_TYPES.TEMPLATE_DISCOUNT`
  `if (age > 18)` => `if (age > AGE_ADULTHOOD)`

## Modularization and code reuse.

It is important to modularize the code into reusable modules or components to avoid code repetition and improve the readability and maintainability of the project.

- [ ] Separate presentation logic and data logic into different components in React.
- [ ] Do not use `useEffect` inside the component but move it to a custom hook either inside or outside the file in question.
- [ ] Use utility functions or external libraries to avoid code repetition
- [ ] Avoid using impure functions.

## Documentation and comments.

- [ ] Write clear and concise comments to explain the **why** and NOT the **what** of code sections, only when really necessary. _In general lines, a good nomenclature should be enough to understand the code._

## Automation tests.

- [ ] Use the real store when developing tests as much as possible (avoid using `'redux-mock-store'` ).
- [ ] Write tests for all functions and components of the project.

lucesitaliss committed:
banner formation

lucesitaliss committed:
Boddy The Page Home

lucesitaliss committed:
body structure for mobile

lucesitaliss committed:
homepage design for mobile

lucesitaliss committed:
home page, mobile

lucesitaliss committed:
chance stroke-width for strokeWidth

lucesitaliss committed:
chance stroke-width for strokeWidth again

lucesitaliss committed:
compatibility solution svg commands

lucesitaliss committed:
adjustment of the height measurements of the home page

lucesitaliss committed:
responsive menu language for mobile

lucesitaliss committed:
change of header design

lucesitaliss committed:
Banner settings

lucesitaliss committed:
desing responsive HomePage

lucesitaliss committed:
banner text translation

lucesitaliss committed:
translation of homepage texts

lucesitaliss committed:
modification of the tests, due to changes

lucesitaliss committed:
footer translation and responsive arrangements

lucesitaliss committed:
line one of the moving banner

lucesitaliss committed:
banner movement lines

lucesitaliss committed:
Footer movement lines

lucesitaliss committed:
adjustments in mobile lines Banner

lucesitaliss committed:
menuLanguageButton responsive

lucesitaliss committed:
banner responsive

lucesitaliss committed:
change Logo color

lucesitaliss committed:
banner responsive

lucesitaliss committed:
responsive content

lucesitaliss committed:
Page Home responsive

lucesitaliss committed:
auto_awesome
Quizás quisiste decir: arreglo de configuración del header
​Ver detalles
35 / 5.000
Resultados de traducción
Resultado de traducción

lucesitaliss committed:
correction in the test

lucesitaliss committed:
banner responsive fix

lucesitaliss committed:
Desing responsive very small screen

lucesitaliss committed:
header responsive

lucesitaliss committed:
bug fix

lucesitaliss committed:
Banner style file creation

lucesitaliss committed:
content Responsive

lucesitaliss committed:
Footer responsive

larts85 commented on src/components/Header/Logo/index.tsx:
```diff
@@ -28,10 +28,10 @@ const Logo: FC<LogoProps> = ({ variant = VARIANT.PRIMARY }) => {
   const colorName = logoColor()
   return (
-    <div className="p-2 flex justify-start gap-2">
+    <div className="flex justify-start gap-[2px] items-center px-[1vw]">
```
porque esto en string inline y no variabilizado como las otras clases?

larts85 commented on src/components/Home/Banner/index.tsx:
```diff
@@ -0,0 +1,41 @@
+      <div id="content" className={content}>
+        <ImageOne />
+        <div id="titleAndSlogan" className={titleAndSlogan}>
+          {translations.bannerHomePage.title}
```
aca faltan esto estilos
```css
    font-size: 75px;
    font-weight: lighter;
    line-height: 128px;
```

ademas todo texto debe estar dentro de una etiqueta de texto, no en un div

larts85 commented on src/components/Home/Banner/index.tsx:
```diff
@@ -0,0 +1,41 @@
+    <div id="banerContainer" className={bannerContainer}>
+      <div id="content" className={content}>
+        <ImageOne />
+        <div id="titleAndSlogan" className={titleAndSlogan}>
```
para que es este id?

lucesitaliss replied:
Para separar el contenido del background y las líneas con movimiento

larts85 commented on src/components/Home/Banner/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        </div>
+        <ImageTwo />
+      </div>
+      <div id="backgound" className={backgound}>
```
porque le pones id a todo, no se hace eso

larts85 commented on src/components/Header/Logo/logo.style.ts:
```diff
@@ -0,0 +1,12 @@
+  sm: 'sm:text-[12px]',
+  md: 'md:text-[13px]',
+  lg: 'lg:text-[15px]',
+  xl: 'xl:text-[20px]',
```
el tamaño maximo esta por debado del especificado en figma `32px`, tanto en logo como en links del navbar, como en subtitle del hero

larts85 commented on src/components/Header/Logo/logo.style.ts:
```diff
@@ -0,0 +1,12 @@
+const size = {
+  movile: 'text-[7px]',
```
el tamaño de los textos en mobile es ilegible, entre desde el celu y por gusto

larts85 commented on src/components/Header/Navbar/navbar.styles.ts:
```diff
@@ -0,0 +1,14 @@
+const size = {
+  movile: 'text-[6.6px]',
```
movile => mobile

larts85 commented on src/components/Header/head.styles.ts:
```diff
@@ -1,11 +1,10 @@
+const size = {
+  movile: 'p-[5px]',
+  sm: 'sm:p-[9px]',
```
estas abusando del los tamaños custom creo, ademas muy pequeños en mobile

larts85 replied:
el nombre de este archivo esta mal

larts85 commented on src/components/Header/index.tsx:
```diff
@@ -1,46 +1,18 @@
-        </div>
-      </div>
+    <header className={hederContainer}>
+      <Logo variant={VARIANT.PRIMARY} data-testid="logo-component" />
```
muy bien modularizando

larts85 commented on src/components/Home/Banner/Background.tsx:
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react'
+
```
Reemplaza todo lo que diga banner por hero

larts85 commented on src/components/Home/Banner/Background.tsx:
```diff
@@ -0,0 +1,21 @@
+const Background: FC = () => {
+  return (
+    <div className="relative">
+      <svg
```
este svg vamos a modificarlo

larts85 commented on src/components/Home/Banner/LineOne.tsx:
```diff
@@ -0,0 +1,33 @@
+import { FC } from 'react'
+
+const LineOneAnimate: FC = () => {
```
adjetivo siempre primero, ademas en participio: `AnimatedLineOne`

larts85 commented on src/components/Home/Banner/LineOne.tsx:
```diff
@@ -0,0 +1,33 @@
+
+const LineOneAnimate: FC = () => {
+  return (
+    <div className="relative top-[-43.2vw]">
```
ese top negativo y usando el ancho del viewport es raro. Explicamelo 🙏🏼

larts85 commented on src/components/Home/Banner/LineTwo.tsx:
```diff
@@ -0,0 +1,33 @@
+import { FC } from 'react'
+
+const LineTwo: FC = () => {
```
no se corresponde este nombre con el de la otra linea, tampoco con los nombres de archivos, uniformidad

larts85 commented on src/components/Home/Content/content.styles.ts:
```diff
@@ -0,0 +1,40 @@
+const className = {
+  content: 'relative',
+  contentCotainer: 'z-20 h-[100vh] w-[100vw] items-center flex flex-col',
+  sesionOne: `font-jost text-right font-normal leading-tight relative top-[31vw] right-[15vw] ${sesionOneSize.movile} ${sesionOneSize.sm} ${sesionOneSize.md} ${sesionOneSize.lg} ${sesionOneSize.xl}`,
```
reemplaza movile por mobile en todo el repo

larts85 commented on src/components/Home/Content/content.styles.ts:
```diff
@@ -0,0 +1,40 @@
+  sesionTwo: `relative left-[15vw] top-[80vw]`,
+  title: `flex flex-col font-jost text-center font-bold pb-[7vw] ${titleSize.movile} ${titleSize.sm} ${titleSize.md} ${titleSize.lg} ${titleSize.xl}`,
+  list: `flex-shrink-0 relative text-left font-normal leading-tight overflow-hidden text-ellipsis ${listSize.movile} ${listSize.sm} ${listSize.md} ${listSize.lg} ${listSize.xl}`,
+  sesionThree: `relative top-[125vw]  md:top-[135vw] right-[15vw] flex-shrink-0 font-jost text-left font-normal leading-tight ${sesionTreeSize.movile} ${sesionTreeSize.sm} ${sesionTreeSize.md} ${sesionTreeSize.lg} ${sesionTreeSize.xl}`,
```
sesion por session

larts85 commented on src/components/Header/Navbar/navbar.styles.ts:
```diff
@@ -0,0 +1,14 @@
+
+const className = {
+  container: 'sm:absolute sm:left-1/2 sm:-translate-x-1/2',
+  ul: 'flex flex-row items-center gap-[3vw] px-[1vw]',
```
ese gap no deberia depender del ancho del viewport, queda raro, sigues sin reproducir al detalle el figma

larts85 commented on src/components/Header/index.tsx:
```diff
@@ -1,46 +1,18 @@
-          <LanguagesButton />
-        </div>
-      </div>
+    <header className={hederContainer}>
```
el header deberia tener un `max-width`, esto te lo dije hace rato, te explique que la pagina y los elementos no pueden espaciarse lateralmente hasta el infinito, el diseño del figma esta definido para `1440px`, ese es tu `max-width`

larts85 commented on src/components/Home/Banner/banner.styles.ts:
```diff
@@ -0,0 +1,25 @@
+const className = {
+  bannerContainer: 'relative flex flex-col',
+  content: 'flex relative z-20 justify-around',
+  titleAndSlogan: `absolute inset-0 top-[-23vw] flex flex-col items-center justify-center font-cormorant text-neutral-darker whitespace-nowrap ${titleAndSloganSize.movile} ${titleAndSloganSize.sm} ${titleAndSloganSize.md} ${titleAndSloganSize.lg} ${titleAndSloganSize.xl}`,
```
estos tamaños no se corresponden con el figma, tampoco el padding del recuadro del slogan, ni su ancho de borde, tienes que calcar el diseño, no es mas o menos, se espera que desarrolles [pixel perfect](https://medium.com/@its.varavex/why-pixel-perfect-design-ebcf53d4f276)

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,54 @@
+          {translations.contentHomePage.sesionOne.map((sesion) => (
+            <p key={sesion}>
+              {sesion}
+              <br></br>
```
estos saltos de linea no deberian usarse, hazlo con css, ademas esta mal logrado el espaciado de los textos, dependiendo de la pantalla, tienes que encontrarle otro enfoque me parece. Los tamaños de los textos tampoco estan bien.

larts85 commented on src/components/Home/Footer/FooterWhatsapp.tsx:
```diff
@@ -0,0 +1,25 @@
+
+const FooterWhatsapp: FC = () => {
+  return (
+    <div className="absolute top-[18vw] left-[86vw]">
```
no entiendo porque necesitas envolver cada `svg` dentro de un `div`, creo que no es necesario, se le puede pasar la className al svg, y ponerlos todos en una carpeta `/components/images/...`

larts85 commented on src/components/Home/Footer/index.tsx:
```diff
@@ -0,0 +1,29 @@
+      <FooterLineThree />
+      <FooterWhatsapp />
+
+      <div id="copyrightContainer" className={className.copyrightContainer}>
```
de vuelta estos ids...

larts85 commented on src/components/LanguagesButton/index.test.tsx:
```diff
@@ -25,18 +25,18 @@ describe('LanguagesButon', () => {
 
-  it('should have a mobile layout if rendered with: isMobile', () => {
-    render(<LanguagesButton isMobile />)
+  // it('should have a mobile layout if rendered with: isMobile', () => {
```
porque comentado?

larts85 commented on src/components/LanguagesButton/languagesButtonStyles.ts:
```diff
@@ -1,14 +1,30 @@
+const containerSize = {
```
me gusta este sistema de evidenciar/declarar los breakpoints

larts85 commented on src/pages/index.tsx:
```diff
@@ -15,11 +18,12 @@ const Home: NextPage = () => {
-      <button className="btn-primary">Save changes</button>
+      <Header />
+      <main>
+        <Banner />
```
cada componente dentro de main deberia tener como elemento primario un `<section>` no un `<div>`

larts85 commented on tailwind.config.js:
```diff
@@ -41,6 +41,7 @@ module.exports = {
       },
       fontFamily: {
         jost: ['Jost', 'sans-serif'],
+        cormorant: ['Cormorant Garamond', 'serif', 'normal'],
```
investiga porque no se compaginan las font de figma con el navegador

larts85 commented on tsconfig.json:
```diff
@@ -16,6 +16,7 @@
     "incremental": true,
     "baseUrl": ".",
     "paths": {
+      // "react": ["./node_modules/@types/react"],
```
porque lo agregas comentado?, si no se requiere sacalo

lucesitaliss committed:
Breakpoint xxl is added to the Header

lucesitaliss committed:
Screen size 1440 is added

lucesitaliss committed:
Home settings

lucesitaliss committed:
change footer

lucesitaliss committed:
horizontal overflow arrangement and other details

lucesitaliss committed:
Banner and Content settings

lucesitaliss committed:
add footer

lucesitaliss committed:
responsive adjustment in Content

lucesitaliss committed:
footer responsive

lucesitaliss committed:
Home Page Responsive

lucesitaliss committed:
Home Page Responsive

lucesitaliss committed:
double space removal

lucesitaliss committed:
footer route change

larts85 commented on src/components/Footer/FooterBackground.tsx:
```diff
@@ -0,0 +1,22 @@
+
+const FooterBackground: FC = () => {
+  return (
+    <div className="absolute top-[-2vw] xxl:top-[-20px] -z-10 w-full">
```
no entiendo porque una posición del eje y, como top, recibe un valor de ancho del viewport y no del alto `vh`?

larts85 commented on src/components/Footer/FooterLineOne.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const FooterLineOne: FC = () => {
+  return (
+    <div className="absolute z-20 top-[-9vw] xxl:top-[-110px]">
```
misma pregunta

larts85 commented on src/components/Footer/FooterLineThree.tsx:
```diff
@@ -0,0 +1,33 @@
+
+const FooterLineThree: FC = () => {
+  return (
+    <div className="absolute z-20 top-[-15vw] xxl:top-[-230px]">
```
x3

larts85 commented on src/components/Footer/FooterLineTwo.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const FooterLineTwo: FC = () => {
+  return (
+    <div className="absolute z-20 top-[-20vw] xxl:top-[-300px]">
```
x4

larts85 commented on src/components/Footer/FooterWhatsapp.tsx:
```diff
@@ -0,0 +1,24 @@
+const FooterWhatsapp: FC = () => {
+  return (
+    <svg
+      width="10vw"
```
el ancho de un icono no debería depender del ancho de la pantalla

lucesitaliss replied:
Una buena práctica para hacer que un logo SVG sea responsive es utilizar medidas relativas en lugar de medidas absolutas. Aquí hay algunos pasos que puedes seguir:
Usar unidades relativas: En lugar de especificar un tamaño fijo para el logo SVG, puedes usar unidades relativas como porcentaje (%) o viewport units (vw, vh) para que el tamaño se ajuste de manera proporcional al contenedor o al tamaño de la pantalla.

Viewport units (vw, vh): Puedes usar las unidades vw y vh para especificar el tamaño del logo en relación con el tamaño de la ventana gráfica. Por ejemplo, width: 10vw; height: 10vh; hará que el logo ocupe el 10% del ancho de la pantalla y el 10% de su altura.

Media queries: Utiliza media queries en tu CSS para establecer diferentes tamaños del logo dependiendo del tamaño de la pantalla. Por ejemplo, puedes definir diferentes tamaños para dispositivos móviles, tabletas y pantallas de escritorio.

Contenedor flexible: Coloca el logo dentro de un contenedor flexible, como un div con display: flex; o display: grid;, y utiliza propiedades como justify-content y align-items para centrar y ajustar el tamaño del logo dentro del contenedor flexible.

Aquí tienes un ejemplo de cómo podrías aplicar estas prácticas en tu código CSS:

larts85 commented on src/components/Footer/footer.styles.ts:
```diff
@@ -0,0 +1,22 @@
+const copyrightSize = {
```
yo creo que hay algunas de estas medidas que coinciden con las medidas de tailwind, no deberias usar todas custom todo el tiempo, la idea es que sea en casos especiales

larts85 commented on src/components/Footer/footer.styles.ts:
```diff
@@ -0,0 +1,22 @@
+    'flex flex-col items-end justify-end h-[29vw] lg:h-[25vw] xxl:h-[26vw] max-h-[471px] xlg:h-[385px]',
+  whatsapp: 'pr-[3vw] pb-[3vw] xlg:pr-[40px] xlg:pb-[20px]',
+  copyrightContainer:
+    'bg-secondary-darker w-[100vw] h-[7vw] flex items-center justify-center max-w-[1440px] max-h-[100.8px]',
```
aca tambien asocias el `height` con el `vw`

larts85 commented on src/components/Header/Logo/index.tsx:
```diff
@@ -21,21 +22,16 @@ const Logo: FC<LogoProps> = ({ variant = VARIANT.PRIMARY }) => {
       <svg
-        width="73"
-        height="43"
+        width="4vw"
```
no usar vw

larts85 commented on src/components/Header/Logo/index.tsx:
```diff
@@ -1,6 +1,7 @@
+import className from './logo.style'
 
 const Logo: FC<LogoProps> = ({ variant = VARIANT.PRIMARY }) => {
   const logoStyleVariants: StyleVariants = {
```
sacar afuera del componente

larts85 commented on src/components/Header/Logo/index.tsx:
```diff
@@ -21,21 +22,16 @@ const Logo: FC<LogoProps> = ({ variant = VARIANT.PRIMARY }) => {
         role="svg"
         fill="none"
         data-testid="logo"
+        className="max-w-[72.4528px]"
```
??.????

larts85 commented on src/components/Header/Navbar/navbar.styles.ts:
```diff
@@ -0,0 +1,14 @@
+const size = {
```
no usar todos custom

larts85 commented on src/components/Header/head.types.ts:
```diff
@@ -0,0 +1,3 @@
+export interface headProp {
```
este nombre esta mal, no es head, y no es prop en singular

larts85 commented on src/components/Header/index.tsx:
```diff
@@ -1,45 +1,28 @@
+const { headerContainer, navAndButton, headercontent } = className
 
-const Header: FC = () => {
+const Header: FC<headProp> = ({ transparent }) => {
```
ademas los tipos se declaran con mayuculas

larts85 commented on src/components/Home/Content/ContentBackground.tsx:
```diff
@@ -0,0 +1,65 @@
+import { FC } from 'react'
+
+const ContentBackground: FC = () => {
```
inviertir, en ingles adjetico primero

larts85 commented on src/components/Home/Content/content.styles.ts:
```diff
@@ -0,0 +1,56 @@
+  xxl: 'xxl:text[32px] xxl:pl-[6vw]',
+  xlg: 'xlg:pl-[80px] ',
+}
+const sesionTreeTextSize = {
```
ortografía

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+import { FC } from 'react'
+import ContentBackground from './ContentBackground'
+import ContentImageOne from './ContentImageOne'
+import ContentImageTwo from './ContentImageTwo'
```
las imágenes svg deberían estar en una carpeta

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+            {translations.contentHomePage.sesionOne.map((sesion) => (
+              <p key={sesion}>
+                {sesion}
+                <br></br>
```
reemplazar estos saltos de linea por estilos

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+  const { translations } = useTranslations()
+
+  return (
+    <div className={className.mainConter}>
```
ortografia

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+
+  return (
+    <div className={className.mainConter}>
+      <div className={className.contentBackground}>
```
este div creo que es innecesario, pero si realmente lo fuese debería estar dentro del componente

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+
+      <div className={className.contentCotainer}>
+        <div className={className.sessionOne}>
+          <div className={className.textsesionOne}>
```
falta de ortografia

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+      </div>
+
+      <div className={className.contentCotainer}>
+        <div className={className.sessionOne}>
```
innecesario, anidar 2 div es incorrecto, 3 es demasiado

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+        </div>
+        <div className={className.sesionTwo}>
+          <ContentImageTwo />
+          <div className={className.sesionTwoText}>
```
innecesario, session va con 2 ss, es repetitivo este error

larts85 commented on src/components/Home/Content/index.tsx:
```diff
@@ -0,0 +1,64 @@
+            </ul>
+          </div>
+        </div>
+        <div className={className.sesionThree}>
```
...

larts85 commented on src/components/LanguagesButton/IconsComponents/Arrow.tsx:
```diff
@@ -0,0 +1,30 @@
+const MenuArrow = () => {
+  return (
+    <div className="max-w-[12.17px] max-h-[8.05px] w-[3px] sm:w-[5px] md:w-[6px] lg:w-[7px] xl:w-[10px] xxl:w-[12.17px]">
```
no creo que en los px las centenas hagan diferencia, no se usan decimales en los px

larts85 commented on src/components/LanguagesButton/LanguagesMenu/index.tsx:
```diff
@@ -21,33 +21,26 @@ const LanguagesMenu: FC<LangMenuProps> = ({ isMobile = false }) => {
-                }
+                className={`${linkMenu}
+                  ${
+                    currentLang === lang
```
en este ternario podrias condicionar solo la palabra que cambia que es la del medio

larts85 commented on tailwind.config.js:
```diff
@@ -9,6 +9,10 @@ module.exports = {
     extend: {
+      screens: {
+        xxl: '1440px',
+        xlg: '1441px',
```
porque este xlg con solo un px mas, innecesario

larts85 reviewed (changes requested):
elimina la mayoria de las medidas custom `[...]`, hay muchos seguro qya tailwind soporta, y los que se repitan los pones en la config.

larts85 commented:
![image](https://github.com/larts85/masaje_las_manos/assets/68341136/af332b5d-2cd1-4571-9721-a227f6b0f83f)

no debería verse blanco por fuera ❌, ademas se nota el espacio entre el hero y main como se ve en el cuadrado rojo

lucesitaliss committed:
Style configuration, banner layout change

lucesitaliss committed:
layout is passed to grid

lucesitaliss committed:
content session 3 setting

lucesitaliss committed:
auto_awesome
Traducir del: inglés
​
17 / 5.000
Resultados de traducción
Resultado de traducción

lucesitaliss committed:
home responsive

lucesitaliss committed:
update grid

lucesitaliss committed:
Home page
